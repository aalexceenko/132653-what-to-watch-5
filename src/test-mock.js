import {getFilmRank} from "./utils";

export const VISIBLE_FILMS_COUNT = 8;
export const ID = 1;
export const VIDEO_CURRENT_TIME = 0;
export const PROGRESS_POSITION = 0;
export const AUTHORIZATION_STATUS = `AUTH`;

export const IS_VIDEO = true;
export const IS_PLAYING = true;
export const IS_IN_MY_LIST = false;
export const TAB = `details`;
export const PREVIEW = `img/aviator.jpg`;
export const PREVIEW_VIDEO = `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
export const TITLE = `Bohemian Rhapsody`;
export const ACTIVE_GENRE = `Comedy`;
export const GENRES = [`Comedy`, `Crime`, `Documentary`, `Drama`, `Horror`, `Kids & Family`];
export const MATCH = {
  params: {
    id: 1
  }
};

export const FILMS = [
  {
    description: `A young man who was sentenced to seven years`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 1,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    year: 2008,
    ratingText: getFilmRank(3.6),
    myList: false,
    comments: false,
  },
  {
    description: `A young man who was`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 2,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    year: 2008,
    ratingText: getFilmRank(3.6),
    myList: false,
    comments: false,
  },
];

export const FILM = {
  description: `A young man who was sentenced to seven years`,
  rating: 3.6,
  director: `Nicolas Winding Refn`,
  genre: `Action`,
  id: 3,
  title: `Bronson`,
  actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  votes: 109661,
  runtime: 92,
  year: 2008,
  ratingText: getFilmRank(3.6),
  myList: false,
  comments: false,
};

export const REVIEWS = [
  {
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    user: {
      name: `Christina`,
      id: 1,
    },
    comment: `This film is an experience and i has already seen it 4 times`,
  },
  {
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    user: {
      name: `Christina`,
      id: 2,
    },
    comment: `This film is an experience and i has already seen it 4 times`,
  }
];
