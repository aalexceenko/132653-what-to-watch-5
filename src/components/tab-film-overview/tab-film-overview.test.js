import React from "react";
import renderer from "react-test-renderer";
import TabFilmOverview from "./tab-film-overview";
import {FILM} from "../../test-mock";

it(`Should TabFilmOverview render correctly`, () => {
  const tree = renderer
    .create(
        <TabFilmOverview
          description={FILM.description}
          rating={FILM.rating}
          ratingText={FILM.ratingText}
          votes={FILM.votes}
          actors={FILM.actors}
          director={FILM.director}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
