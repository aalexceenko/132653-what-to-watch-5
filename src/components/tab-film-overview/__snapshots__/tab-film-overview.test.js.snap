// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should TabFilmOverview render correctly 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      3.6
       
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Normal
         
      </span>
      <span
        className="movie-rating__count"
      >
        109661
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      A young man who was sentenced to seven years
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Nicolas Winding Refn
         
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Tom Hardy, Kelly Adams, Luing Andrews
         
      </strong>
    </p>
  </div>,
]
`;
